src/main/java/com/kodekonveyor/market/project/UpdateProjectModelController.java

```diff
@@ -43,22 +57,59 @@
 
     validateAuthoization(project);
 
-    final Set<Long> milestoneIds = projectModelDTO.getMilestone();
+    final Set<MilestoneDTO> milestones = projectModelDTO.getMilestone();
+    final Set<Long> milestonIds = new HashSet<>();
+    milestones.forEach((milestone) -> milestonIds.add(milestone.getId()));
     project.setMilestone(
         Sets.newHashSet(
             milestoneEntityRepository
-                .findAllById(milestoneIds)
+                .findAllById(milestonIds)
         )
     );
-    ProjectEntity projectEntityUpdated = projectEntityRepository.save(project);
+
+    updateTasks(projectModelDTO);
+
+    final ProjectEntity projectEntityUpdated =
+        projectEntityRepository.save(project);
+
     logger.debug(
         LoggingMarkerConstants.PROJECT,
         ProjectConstants.PROJECT_DTO_RETURNED_SUCCESSFULLY + project.getId()
     );
+
     return getProjectDTO(projectEntityUpdated);
 
   }
 
+  private void updateTasks(
+      final ProjectModelDTO projectModelDTO
+  ) {
+    final Set<TaskDTO> projectDTOTasks = projectModelDTO.getTask();
+    final List<TaskEntity> tasks = convertTaskDTOs(projectDTOTasks);
+    for (final TaskEntity task : tasks) {
+      final TaskEntity updatedTask = updateTasksService.call(task);
+      taskEntityRepository.save(updatedTask);
+    }
+  }
+
+  private List<TaskEntity> convertTaskDTOs(final Set<TaskDTO> projectDTOTasks) {
+    final TaskEntity task = new TaskEntity();
```

Comment:
You are updating the same task entity and add this entity to your array many times.
Create a convert method that creates a new entity for a single TaskDTO